handlers/auth_handler.go

```diff
@@ -472,3 +472,237 @@ func LogoutHandler(c echo.Context) error {
 	logger.Infof("User logged out successfully")
 	return c.NoContent(http.StatusNoContent)
 }
+
+// ForgotPasswordHandler godoc
+// @Summary      Request password reset
+// @Description  Sends a password reset email to the user's registered email address
+// @Tags         auth
+// @Accept       json
+// @Produce      json
+// @Param        forgotPasswordRequest  body  ForgotPasswordRequest  true  "Forgot password request"
+// @Success      200 {object} GenericResponse "Password reset email sent successfully"
+// @Failure      400 {object} echo.HTTPError  "Bad request"
+// @Failure      404 {object} echo.HTTPError  "User not found"
+// @Failure      429 {object} echo.HTTPError  "Too many requests"
+// @Failure      500 {object} echo.HTTPError  "Internal server error"
+// @Router       /v1/auth/forgot-password [post]
+func ForgotPasswordHandler(c echo.Context) error {
+	logger := c.Logger()
+
+	var req ForgotPasswordRequest
+	if err := c.Bind(&req); err != nil {
+		logger.Error("Invalid forgot password request payload:", err)
+		return echo.ErrBadRequest
+	}
+
+	if req.Email == "" {
+		logger.Error("Email is required.")
+		return &echo.HTTPError{
+			Code:    http.StatusBadRequest,
+			Message: "email field is required",
+		}
+	}
+
+	newCrypto := crypto.NewCrypto()
+	user := models.User{}
+
+	emailPseudo, err := newCrypto.HashData([]byte(req.Email), "HMAC-SHA-256")
+	if err != nil {
+		logger.Errorf("Failed to hash email: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	if err := db.Conn.Where("email_pseudonym = ?", emailPseudo).First(&user).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			logger.Error("User not found for password reset.")
+			return c.JSON(http.StatusOK, GenericResponse{
+				Message: "If an account with that email exists, we've sent a password reset link",
+			})
+		}
+
+		logger.Errorf("Failed to find user: %v", err)
+		return echo.ErrInternalServerError
+	}
+
+	recentReset := models.PasswordReset{}
```

Comment:
No need to create a `models.EmailVerification{}`. We can use `models.EmailVerification{}` since it has similar model attributes to `models.EmailVerification{}` already.